app/src/main/java/com/gmail/mxwild/newhabit/movieslist/FragmentMoviesList.kt

```diff
@@ -30,32 +33,47 @@ class FragmentMoviesList : Fragment() {
         val recycler: RecyclerView = view.findViewById(R.id.movie_list)
         adapter = MoviesAdaptor(clickListener)
         recycler.adapter = adapter
-    }
 
-    override fun onStart() {
-        super.onStart()
-        loadMovies()
-    }
+        observeMovies()
 
-    private fun loadMovies() {
-
-        viewLifecycleOwner.lifecycleScope.launch {
+        if (viewModel.moviesList.value.isNullOrEmpty()) {
```

Comment:
This check should be part of view model and not here

Reply:
You right